in-person-payments-example/src/main/java/com/adyen/ipp/web/InPersonPaymentsWebController.java

```diff
@@ -0,0 +1,99 @@
+package com.adyen.ipp.web;
+
+import com.adyen.ipp.ApplicationProperty;
+import com.adyen.ipp.model.Table;
+import com.adyen.ipp.service.PosTransactionStatusService;
+import com.adyen.ipp.service.TableService;
+import com.adyen.service.exception.ApiException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import java.io.IOException;
+
+@Controller
+public class InPersonPaymentsWebController {
+
+    private final Logger log = LoggerFactory.getLogger(InPersonPaymentsWebController.class);
+
+    @Autowired
+    private TableService tableService;
+
+    @Autowired
+    private PosTransactionStatusService posTransactionStatusService;
+
+    @Autowired
+    public InPersonPaymentsWebController(ApplicationProperty applicationProperty) {
+        this.applicationProperty = applicationProperty;
+    }
+
+    @Autowired
+    private ApplicationProperty applicationProperty;
```

Comment:
I don't think you need to define the dependency here as it is injected in the constructor.

I see `ApplicationProperty` injected in the constructor wherever it is needed (controllers, services). This approach is fine of course (semantically correct), but I personally would prefer a dependency (setter/getter) injection that gives a default constructor with more flexibility (in case of testing).

Both approaches are fine IMO so I think it is matter of personal choice.